core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -98,6 +98,12 @@ sealed abstract class ZLayer[-RIn, +E, +ROut] { self =>
   )(implicit tag: EnvironmentTag[ROut2]): ZLayer[RIn with RIn2, E1, ROut1 with ROut2] =
     self.++[E1, RIn2, ROut1, ROut2](that)
 
+  @deprecated("For macros only. Use ++ or and")
+  final def andEager[E1 >: E, RIn2, ROut1 >: ROut, ROut2](
+    that: ZLayer[RIn2, E1, ROut2]
+  ): ZLayer[RIn with RIn2, E1, ROut1 with ROut2] =
+    self.zipWithPar(that)(_.unionAll[ROut2](_))
```

Comment:
See [this comment](https://github.com/zio/zio/issues/8855#issuecomment-2131259888) why I think using `unionAll` is a bad idea.

Can you check whether my suggestion [here](https://github.com/zio/zio/issues/8855#issuecomment-2129512781) works?

Reply:
Ahhh I just saw that you added pruning as part of the macro - please disregard